Common/Orders/TimeInForces/DayTimeInForceHandler.cs

```diff
@@ -0,0 +1,64 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using QuantConnect.Interfaces;
+using QuantConnect.Securities;
+
+namespace QuantConnect.Orders.TimeInForces
+{
+    /// <summary>
+    /// Handles the Day time in force for an order (DAY)
+    /// </summary>
+    public class DayTimeInForceHandler : ITimeInForceHandler
+    {
+        private readonly IAlgorithm _algorithm;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DayTimeInForceHandler"/> class
+        /// </summary>
+        /// <param name="algorithm">The instance of the algorithm</param>
+        public DayTimeInForceHandler(IAlgorithm algorithm)
+        {
+            _algorithm = algorithm;
+        }
+
+        /// <summary>
+        /// Checks if an order has expired
+        /// </summary>
+        /// <param name="order">The order to be checked</param>
+        /// <returns>Returns true if the order has expired, false otherwise</returns>
+        public bool HasOrderExpired(Order order)
+        {
+            var exchangeHours = MarketHoursDatabase
+                .FromDataFolder()
+                .GetExchangeHours(order.Symbol.ID.Market, order.Symbol, order.Symbol.SecurityType);
+
+            var time = _algorithm.UtcTime.ConvertFromUtc(exchangeHours.TimeZone);
+
+            return !exchangeHours.IsOpen(time, false);
```

Comment:
Is this correct? Isn't it possible to move to the next day and have `IsOpen` return true again. Is it more correct to compare the order's submit date and the current date (in the security's time zone) ?

What happens with 24 hour markets? If I say time in force = day at 23:59PM - does it cancel at the stroke of midnight?

Reply:
Good point, this might need to be different per security type:
- Forex: both FXCM and Oanda use 5 PM New York time as day close
- Crypto: GDAX does not support Day, so maybe we should use midnight UTC here?